tftypes/attribute_path.go

```diff
@@ -149,7 +149,16 @@ func (a *AttributePath) WithElementKeyString(key string) *AttributePath {
 
 // WithElementKeyInt adds an ElementKeyInt step to `a`, using `key` as the
 // element's key. `a` is copied, not modified.
-func (a *AttributePath) WithElementKeyInt(key int64) *AttributePath {
+func (a *AttributePath) WithElementKeyInt(key int) *AttributePath {
+	steps := a.Steps()
+	return &AttributePath{
+		steps: append(steps, ElementKeyInt(key)),
+	}
+}
+
+// WithElementKeyInt64 adds an ElementKeyInt step to `a`, using `key` as the
+// element's key. `a` is copied, not modified.
+func (a *AttributePath) WithElementKeyInt64(key int64) *AttributePath {
```

Comment:
My controversial hot take is we should leave this out for now, and wait for someone to need it.

Reply:
I'm also completely fine with that. Wasn't sure the boundaries of fully offering "protocol support" (64 bit), but seems pragmatic to drop this here for practical reasons.

Reply:
I'm generally in the "let's make sure we can support everything the protocol can" camp for this, but I imagine if we have 2 billion elements in a list, something else is going to break first. `cty` _also_ uses an `int`, so it would have problems too. So I'm grudgingly willing to bend to pragmatism on this one. (I also think _Go_ will have some trouble with those values, because I don't know that slices can have more values than `int`s can hold.)

Reply:
Yeah, the Go spec doesn't support Array or Slice types above `int`: https://golang.org/ref/spec#Array_types

> The length is part of the array's type; it must evaluate to a non-negative constant representable by a value of type int.

It would require a different implementation to always support 64 bit "slices" on non 64 bit systems.